store/tikv/coprocessor.go

```diff
@@ -1389,14 +1404,18 @@ func (e *rateLimitAction) broadcastIfNeeded(needed bool) {
 	}
 	e.conditionLock()
 	defer e.conditionUnlock()
-	if !e.cond.exceeded || e.cond.waitingWorkerCnt < 1 {
+	if !e.cond.exceeded {
 		return
 	}
 	for !e.cond.isTokenDestroyed {
 		e.cond.Wait()
 	}
 	e.cond.exceeded = false
 	e.cond.Broadcast()
+	if e.cond.waitingWorkerCnt < 1 && e.cond.consumed {
+		e.cond.consumed = false
+		e.cond.once = sync.Once{}
+	}
```

Comment:
If there is only one worker and 2 token with many tasks (KeepOrder case with 1 concurrency). The OOM may miss being initialized again in the following case:

1. worker handle Task -> cause oom action (consuming once)
2. worker destroyToken directly and continue to work.
3. iterator consmues all data and broadcast it. (set exceeded as false)
4. worker continue working while the `once` in oom action is consumed and never being initialized again.

So we also need to check whether init the once in `broadcastIfNeeded`. As there are 2 places try to init once, I initroduce `consumed` to prevent once from being initialized twice.